### PR TITLE
[enhancement](functioncontext) function context should use shared ptr and simply function context

### PR DESCRIPTION
**Pull request by yiguolei**
# Proposed changes

1. remove unused method from function context;
2. remove mempool from expr context;
3. remove free pool
4. remove unused stringval method, since we will remove stringval in the future.
5. use a shared ptr to save constant col values.
6. rename prepare method to open, it is more meaningful.

## Problem summary

Describe your changes.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [ ] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by yiguolei** [comment]
./run buildall

File: be/src/vec/exprs/vstruct_literal.cpp [hunk 1]
```diff
@@ -25,8 +25,7 @@ Status VStructLiteral::prepare(RuntimeState* state, const RowDescriptor& row_des
     Field struct_field = Tuple();
     for (const auto child : _children) {
         Field item;
-        ColumnPtrWrapper* const_col_wrapper = nullptr;
-        RETURN_IF_ERROR(child->get_const_col(context, &const_col_wrapper));
```
Comment (carlvinhust2012) [inline be/src/vec/exprs/vstruct_literal.cpp:29]: if remove the  RETURN_IF_ERROR，no need to report the error message ?

Comment (yiguolei) [reply be/src/vec/exprs/vstruct_literal.cpp:29]: Thanks. It is a mistake.

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 33.76 seconds
 stream load tsv:          475 seconds loaded 74807831229 Bytes, about 150 MB/s
 stream load json:         39 seconds loaded 2358488459 Bytes, about 57 MB/s
 stream load orc:          67 seconds loaded 1101869774 Bytes, about 15 MB/s
 stream load parquet:          27 seconds loaded 861443392 Bytes, about 30 MB/s
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20230302023347_clickbench_pr_106736.html

**Comment by yiguolei** [comment]
run buildall

**Review by Gabriel39 — APPROVED** [review]
LGTM

**Pull request merged**